### PR TITLE
[DOC] Improvements to Python docs.

### PR DESCRIPTION
**Pull request by rxin**

### CODE REVIEW

**Comment by rxin** [comment]
cc @davies - can you build to make sure this is OK on your computer? Also I added deprecation warning for 2 methods.

File: python/pyspark/sql/context.py [hunk 1]
```diff
@@ -173,31 +173,8 @@ def _inferSchema(self, rdd, samplingRatio=None):
...
-        >>> rdd = sc.parallelize(
-        ...     [Row(field1=1, field2="row1"),
-        ...      Row(field1=2, field2="row2"),
-        ...      Row(field1=3, field2="row3")])
-        >>> df = sqlCtx.inferSchema(rdd)
-        >>> df.collect()[0]
-        Row(field1=1, field2=u'row1')
-        """
+        """DEPRECATED: use :func:`createDataFrame` instead"""
+        warnings.warn("Use createDataFrame instead of inferSchema.", DeprecationWarning)
```
Comment (davies) [inline python/pyspark/sql/context.py]: inferSchema() is called in `createDataFrame`, we should change that.

Comment (rxin) [inline python/pyspark/sql/context.py]: I updated it.

**Comment by rxin** [comment]
@davies I took the deprecation part out of this PR. Can you submit a PR to throw a proper deprecation warning?

**Comment by rxin** [comment]
Merging in master & branch-1.3.

**Comment by davies** [comment]
@rxin will do.

**Comment by rxin** [comment]
Thanks.

**Pull request closed without merging**